fftoml/fftoml.go

```diff
@@ -6,27 +6,69 @@ import (
 	"io"
 	"strconv"
 
-	"github.com/BurntSushi/toml"
+	"github.com/pelletier/go-toml"
 	"github.com/peterbourgon/ff"
 )
 
 // Parser is a parser for TOML file format. Flags and their values are read
 // from the key/value pairs defined in the config file.
 func Parser(r io.Reader, set func(name, value string) error) error {
-	var m map[string]interface{}
-	_, err := toml.DecodeReader(r, &m)
-	if err != nil {
-		return ParseError{Inner: err}
+	return ParserWith()(r, set)
+}
+
+type config struct {
+	separator string
+}
+
+// Option is a function which configures a fftoml.ConfigFileParser
+type Option func(*config)
+
+// FlagSeparator is an option which configures a separator
+// to use when constructing a flag name
+func FlagSeparator(s string) Option {
+	return func(c *config) {
+		c.separator = s
 	}
-	for key, val := range m {
-		values, err := valsToStrs(val)
+}
+
+// ParserWith configures and returns a new ConfigFileParser using the
+// provided slice of Option types
+// By default the returned Parser uses a "." as a separator
+func ParserWith(opts ...Option) ff.ConfigFileParser {
+	return func(r io.Reader, set func(name, value string) error) error {
+		config := config{separator: "."}
```

Comment:
Hmm, what do you think about adding a Parse method to the Config struct, matching the ConfigFileParser signature, and returning that? e.g.

```go
func ParserWith(opts ...Option) ff.ConfigFileParser {
    c := Config{
        separator: ".",
    }
    for _, opt := range opts {
        opt(&c)
    }
    return c.Parse
}
```

Reply:
That is nice. Does it continue to make sense to call in Config? Perhaps there is a more suitable name. Like `ConfigFileParser` to match the `ff.ConfigFileParser` function signature?